### PR TITLE
Core: Validation of Alias GVL IDs

### PR DESCRIPTION
**Pull request by marki1an**
Corresponding ticket: [2126](https://github.com/prebid/prebid-server/pull/2126)
[PB Java Alias VendorID Resolver Code](https://github.com/prebid/prebid-server-java/blob/master/src/main/java/org/prebid/server/privacy/gdpr/VendorIdResolver.java)

### CODE REVIEW

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 1]
```diff
@@ -199,6 +200,28 @@ private void validateCur(List<String> currencies) throws ValidationException {
         }
     }
 
+    private void validateAliasesGVLIDs(ExtRequestPrebid extRequestPrebid,
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/validation/RequestValidator.java:203]: Nitpick: rename to `validateAliasesGvlIds`

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 2]
```diff
@@ -199,6 +200,28 @@ private void validateCur(List<String> currencies) throws ValidationException {
...
+                                       Map<String, String> aliases) throws ValidationException {
+
+        final Map<String, Integer> aliasgvlids = extRequestPrebid.getAliasgvlids() != null
+                ? extRequestPrebid.getAliasgvlids()
+                : Collections.emptyMap();
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/validation/RequestValidator.java:208]: Can be simplified with
```
final Map<String, Integer> aliasGvlIds = MapUtils.emptyIfNull(extRequestPrebid.getAliasgvlids());
```

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 3]
```diff
@@ -199,6 +200,28 @@ private void validateCur(List<String> currencies) throws ValidationException {
...
+
+        for (Map.Entry<String, Integer> aliasgvlid : aliasgvlids.entrySet()) {
+
+            if (!aliases.containsKey(aliasgvlid.getKey())) {
+                throw new ValidationException("request.ext.prebid.aliasgvlids. vendorId %s refers to"
+                        + " unknown bidder alias: %s", aliasgvlid.getKey(), aliasgvlid.getValue());
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/validation/RequestValidator.java:214]: `aliasGvlId.getKey()` and `aliasGvlId.getValue()` are swapped.

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 4]
```diff
@@ -199,6 +200,28 @@ private void validateCur(List<String> currencies) throws ValidationException {
...
+            }
+
+            if (aliasgvlid.getValue() < 1) {
+                throw new ValidationException("request.ext.prebid.aliasgvlids. "
+                        + "Invalid vendorId %s for alias: %s. Choose a different vendorId, or "
+                        + "remove this entry.", aliasgvlid.getKey(), aliasgvlid.getValue());
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/validation/RequestValidator.java:220]: aliasGvlId.getKey() and aliasGvlId.getValue() are swapped.

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 5]
```diff
@@ -199,6 +200,28 @@ private void validateCur(List<String> currencies) throws ValidationException {
...
     }
 
+    private void validateAliasesGVLIDs(ExtRequestPrebid extRequestPrebid,
+                                       Map<String, String> aliases) throws ValidationException {
+
+        final Map<String, Integer> aliasgvlids = extRequestPrebid.getAliasgvlids() != null
+                ? extRequestPrebid.getAliasgvlids()
+                : Collections.emptyMap();
+
+        for (Map.Entry<String, Integer> aliasgvlid : aliasgvlids.entrySet()) {
```
Comment (CTMBNara) [inline src/main/java/org/prebid/server/validation/RequestValidator.java:210]: Nitpick: rename `aliasgvlid ` to `aliasToGvlId`

**Review by CTMBNara — CHANGES_REQUESTED** [review]

**Review by CTMBNara — DISMISSED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**